src/libraries/System.Private.CoreLib/src/System/TypedReference.cs

```diff
@@ -75,13 +75,15 @@ public override bool Equals(object? o)
             throw new NotSupportedException(SR.NotSupported_NYI);
         }
 
-        public static unsafe object ToObject(TypedReference value)
+#if !CORECLR
```

Comment:
You can move it to Mono specific file instead of ifdefing.

Reply:
I thought this method was also used by NAOT?
As in, if I moved this to the Mono file then wouldn't NAOT be left out?

Reply:
From `src\libraries\System.Private.CoreLib\src\System.Private.CoreLib.Shared.projitems`:

```
    <Compile Include="$(MSBuildThisFileDirectory)System\TypedReference.cs" Condition="'$(FeatureNativeAot)' != 'true'" />
```

Reply:
Most of the file depends on reflection internals that are very different between Native and CoreCLR+Mono today. I expect we will be able to enable the shared file for NativeAOT once we get more unified reflection implementation.